crates/llm-remote-http/src/lib.rs

```diff
@@ -0,0 +1,186 @@
+            .headers(headers)
+            .body(body)
+            .send()
+            .await;
+
+        match resp {
+            Ok(resp) => match resp.json::<InferResponseBody>().await {
+                Ok(val) => Ok(wasi_llm::InferencingResult {
+                    text: val.text,
+                    usage: wasi_llm::InferencingUsage {
+                        prompt_token_count: val.usage.prompt_token_count,
+                        generated_token_count: val.usage.generated_token_count,
+                    },
+                }),
+                Err(err) => Err(wasi_llm::Error::RuntimeError(format!(
+                    "Failed to deserialize response: {err}"
+                ))),
+            },
+            Err(err) => Err(wasi_llm::Error::RuntimeError(err.to_string())),
+        }
+    }
+
+    async fn generate_embeddings(
+        &mut self,
+        model: wasi_llm::EmbeddingModel,
+        data: Vec<String>,
+    ) -> Result<wasi_llm::EmbeddingsResult, wasi_llm::Error> {
+        if !self.allowed_models.contains(&model) {
+            return Err(access_denied_error(&model));
+        }
+        let client = self.client.get_or_insert_with(Default::default);
+
+        let mut headers = HeaderMap::new();
+        headers.insert(
+            "authorization",
+            HeaderValue::from_str(&format!("bearer {}", self.auth_token)).map_err(|_| {
+                wasi_llm::Error::RuntimeError("Failed to create authorization header".to_string())
+            })?,
+        );
+
+        let body = serde_json::to_string(&json!({
+            "model": model,
+            "input": data
+        }))
+        .map_err(|_| wasi_llm::Error::RuntimeError("Failed to serialize JSON".to_string()))?;
+
+        let resp = client
+            .request(
+                http::Method::POST,
+                self.url.join("/embed").map_err(|_| {
+                    wasi_llm::Error::RuntimeError("Failed to create URL".to_string())
+                })?,
+            )
+            .headers(headers)
+            .body(body)
+            .send()
+            .await;
+
+        match resp {
+            Ok(resp) => match resp.json::<EmbeddingResponseBody>().await {
+                Ok(val) => Ok(wasi_llm::EmbeddingsResult {
+                    embeddings: val.embeddings,
+                    usage: wasi_llm::EmbeddingsUsage {
+                        prompt_token_count: val.usage.prompt_token_count,
+                    },
+                }),
+                Err(err) => Err(wasi_llm::Error::RuntimeError(format!(
+                    "Failed to deserialize response: {err}"
+                ))),
+            },
+            Err(err) => Err(wasi_llm::Error::RuntimeError(err.to_string())),
+        }
+    }
+}
+
+impl RemoteHttpLlmEngine {
+    pub fn new(url: Url, auth_token: String) -> Self {
+        RemoteHttpLlmEngine {
+            url,
+            auth_token,
+            allowed_models: Default::default(),
+            client: None,
```

Comment:
Why is client an optional value?

Reply:
I was following the pattern in  `outbound-http`